src/Storages/ObjectStorage/StorageObjectStorageCluster.h

```diff
@@ -31,16 +36,81 @@ class StorageObjectStorageCluster : public IStorageCluster
 
     void setClusterNameInSettings(bool cluster_name_in_settings_) { cluster_name_in_settings = cluster_name_in_settings_; }
 
+    String getClusterName(ContextPtr context) const override;
+
+    void setInMemoryMetadata(const StorageInMemoryMetadata & metadata_) override
+    {
+        if (pure_storage)
+            pure_storage->setInMemoryMetadata(metadata_);
+        IStorageCluster::setInMemoryMetadata(metadata_);
```

Comment:
I assume you are calling the method for both simply because it is more convenient and there are no side effects